lib/Service/MailManager.php

```diff
@@ -493,7 +499,30 @@ public function flagMessage(Account $account, string $mailbox, int $uid, string
 			)
 		);
 	}
+	public function tagMessageWithClient(Horde_Imap_Client_Socket $client, Account $account, Mailbox $mailbox, Message $message, Tag $tag, bool $value):void {
 
+		if ($this->isPermflagsEnabled($client, $account, $mailbox->getName()) === true) {
+			try {
+				if ($value) {
+					// imap keywords and flags work the same way
+					$this->imapMessageMapper->addFlag($client, $mailbox, [$message->getUid()], $tag->getImapLabel());
```

Comment:
addFlag / removeFlag accepts a list of uids. 

Do you think we can group the messages by mailbox in MailManager.deleteTag and pass a list of Messages to tagMessageWithClient?

The advantage is that we reduce the number of imap commands.